services/cloudsecuritygroup/securitygroupaws/securitygroupaws.go

```diff
@@ -0,0 +1,183 @@
+	awsSgResourcePath           = "CloudSecurityGroup"
+	awsSgResourceProtectionMode = "protection-mode"
+	awsSgResourceServices       = "services"
+)
+
+// There is a bug when we pass nil inbound or outbound, ticket link: https://dome9-security.atlassian.net/browse/DOME-12727
+type CloudSecurityGroupRequest struct {
+	IsProtected       bool              `json:"isProtected"`
+	SecurityGroupName string            `json:"securityGroupName"`
+	Description       string            `json:"description"`
+	RegionId          string            `json:"RegionId"`
+	CloudAccountId    string            `json:"CloudAccountId"`
+	Services          ServicesRequest   `json:"services"`
+	Tags              map[string]string `json:"tags,omitempty"`
+}
+
+type CloudSecurityGroupResponse struct {
+	SecurityGroupID   int               `json:"securityGroupId"`
+	ExternalID        string            `json:"externalId"`
+	IsProtected       bool              `json:"isProtected"`
+	SecurityGroupName string            `json:"securityGroupName"`
+	Description       string            `json:"description"`
+	VpcID             string            `json:"vpcId"`
+	VpcName           string            `json:"vpcName,omitempty"`
+	RegionName        string            `json:"RegionId"`
+	CloudAccountId    string            `json:"CloudAccountId"`
+	CloudAccountName  string            `json:"cloudAccountName"`
+	Services          ServicesResponse  `json:"services"`
+	Tags              map[string]string `json:"tags"`
+}
+
+type ServicesRequest struct {
+	Inbound  []BoundServicesRequest `json:"inbound"`
+	Outbound []BoundServicesRequest `json:"outbound"`
+}
+
+type ServicesResponse struct {
+	Inbound  []BoundServicesResponse `json:"inbound"`
+	Outbound []BoundServicesResponse `json:"outbound"`
+}
+
+type BoundServicesRequest struct {
+	Name         string  `json:"name"`
+	Description  string  `json:"description"`
+	ProtocolType string  `json:"protocolType"`
+	Port         string  `json:"port"`
+	OpenForAll   bool    `json:"openForAll"`
+	Scope        []Scope `json:"scope"`
+}
+
+type BoundServicesResponse struct {
+	Id           string  `json:"id"`
+	Name         string  `json:"name"`
+	Description  string  `json:"description"`
+	ProtocolType string  `json:"protocolType"`
+	Port         string  `json:"port"`
+	OpenForAll   bool    `json:"openForAll"`
+	Scope        []Scope `json:"scope"`
+	Inbound      bool    `json:"inbound"`
+	ICMPType     string  `json:"icmpType"`
+	ICMPv6Type   string  `json:"icmpv6Type,omitempty"`
+}
+
+type Scope struct {
+	Type string `json:"type"`
+	Data map[string]string   `json:"data"`
+}
+
+type GetSecurityGroupQueryParameters struct {
+	CloudAccountId string
+	RegionId       string
+}
+
+type UpdateProtectionModeQueryParameters struct {
+	ProtectionMode string
+}
+
+// Get will return list of all the security groups in specific region
+func (service *Service) Get(d9CloudAccountID, awsRegionName string) (*[]CloudSecurityGroupResponse, *http.Response, error) {
```

Comment:
Don't we have a Get method to return specific SG? This can be pretty annoying because the provider must have such a method. If API doesn't support such thing, we will have to create some helper function that will pick the desired SG from that returned list